### ISSUE TITLE
Project Licensing

### ISSUE THREAD

**Issue by theorangepotato** [ChapmanRobotics/Self-Driving-Cars#1]
We need to decide very early on in this project what kind of license we want to release this project under. All contributions to this code base will automatically be covered by the chosen license, and any change in license requires the approval of all contributors (unless it is already permitted by the previously chosen license), which can get complicated. This issue will explore the options, and hopefully be resolved quickly.
## Proprietary

This is the sort of "All rights reserved" license that is commonly seen on many works. It is the most restrictive of the options, as it requires any further innovation to be cleared by the copyright holder. The copyright holder would either need to be established as the club or the school, as giving it to each contributor or an individual in the club would create legal chaos and authoritarian control respectively. This is personally my least preferred license.
## GNU General Public License (GPL)

This is what is known as a "copyleft" license. It allows anyone to modify, reuse, and view the code as long as the derivatives are released under the same license. This is a very common license in libre/open source projects, and was made popular through the rise of the GNU/Linux operating system. While this license is the least cooperative with commercial applications, I do not believe that is of much concern, given our purely academic goal.
## Permissive License (MIT, BSD, Apache)

This license is also a libre/open source license, the main difference with the above is that derivative works are not required to be released under the same license. This is the most friendly to later use in a commercial application, however, I personally doubt this project will provide much use to such an endeavor.
## Upstream Licenses

If we decide to use libraries or code that is licensed under the GPL, we will be forced through the legally binding nature of that license to release our entire source code under the GPL. If we use proprietary code, we will have to negotiate our use and license with the respective copyright holder. If we use code released under a permissive license, we can (typically) relicense it however we like. As such, licensing this project under the GPL (v3) is likely to be the easiest option, especially given the prevalence of GPL-licensed libraries and software.
## 

In conclusion, I personally have a strong preference for a free/open source license and a mild preference for the GPL. Other comments, opinions, and questions are of course welcome.
### Resources

[GPL](https://en.wikipedia.org/wiki/GNU_General_Public_License)
[MIT License](https://en.wikipedia.org/wiki/MIT_License)
[BSD License](https://en.wikipedia.org/wiki/BSD_licenses)
[Apache License](https://en.wikipedia.org/wiki/Apache_License)
[Comparison of Free/Open Source Licenses](https://en.wikipedia.org/wiki/Comparison_of_free_and_open-source_software_licenses)
